lib/public/Notification/IHandleDismissedNotification.php

```diff
@@ -0,0 +1,37 @@
+<?php
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2019, Roeland Jago Douma <roeland@famdouma.nl>
+ *
+ * @author Roeland Jago Douma <roeland@famdouma.nl>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCP\Notification;
+
+/**
+ * Interface INotifier classes should implement if they want to process notifications
+ * that are dismissed by the user.
+ *
+ * This can be useful if dismissing the notification will leave it in an incomplete
+ * state. The handler can chose to for example do some default action.
+ */
+interface IHandleDismissedNotification {
```

Comment:
```suggestion
interface IHandleDismissedNotification extends INotifier {
```

Then you can assume an object is both if you do

```php
if ($notifier instanceof IHandleDismissedNotification) {
 ...
}
```

Reply:
(just like https://github.com/nextcloud/server/blob/63a2725f3d38fe8e5e767cd62f65330776435811/lib/public/Authentication/TwoFactorAuth/IActivatableAtLogin.php#L34 and friends)

Reply:
But we can't have more Interfaces in the future, if they are not linear?

Reply:
@nickvergessen well you can if they all extend the base notifier

Reply:
Right, brain fail